Lib/email/utils.py

```diff
@@ -259,21 +259,13 @@ def decode_params(params):
 
     params is a sequence of 2-tuples containing (param name, string value).
     """
-    # Copy params so we don't mess with the original
-    params = params[:]
-    new_params = []
+    new_params = [params[0]]
     # Map parameter's name to a list of continuations.  The values are a
     # 3-tuple of the continuation number, the string value, and a flag
     # specifying whether a particular segment is %-encoded.
     rfc2231_params = {}
-    name, value = params.pop(0)
-    new_params.append((name, value))
-    while params:
-        name, value = params.pop(0)
-        if name.endswith('*'):
-            encoded = True
-        else:
-            encoded = False
+    for name, value in params[1:]:
```

Comment:
True. The real benefit is avoiding repeated `.pop(0)`.

Reply:
But the code can be made more optimal if avoid a copy.

Reply:
It looks to me that the final value of 'encoded' only depends on the final value of name.  If so, the loop is equivalent to
```
if params:
    encoded = params[-1][0].endswith('*')
```
Otherwise, one can skip-iterate without copying with
```
it = iter(params)
try:
    next(it)
except StopIteration:
    pass
for name, value in it:
   ...
```

Reply:
Each iteration of the loop uses the `encoded` value. The `iter` and `next`ing would be a better technique.